Extract embedded secrets and make them configurable (#30)
OpenTermsArchive/deployment: README.md

> See “[Using collections](https://docs.ansible.com/ansible/latest/user_guide/collections_using.html)” in Ansible’s user guide for more information about Ansible collections.

### Versions compatibility

| Deployment | Engine | Federated API |
| --- | --- | --- |

> | 0.1.0 | >= 0.36.0 | >= 0.1.0 |

Reviewer: I am concerned that this is very likely to become outdated. I would feel more comfortable if the dependency was made clear in the dependent (i.e. engine and Federated API) rather than in here.

Author: After IRL discussion, we will remove it for now to avoid outdated documentation

Available [variables](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html) are listed below, along with default values:

| Variable | Description | Default value | Required |
| --- | --- | --- | --- |

> | `ota_engine_ota_bot_private_key` | SSH private key of GitHub user with privileges on snapshots and versions repositories. | No default value | ✔︎ |

Reviewer: Since a private SSH key is usually stored in a file, it's not clear to me if this variable is a path or the file contents 🙂

> | `ota_engine_ota_bot_private_key` | SSH private key of GitHub user with privileges on snapshots and versions repositories. | No default value | ✔︎ |

Reviewer: If this is made configurable, should it still be called “ota bot”? What about
```suggestion
| `ota_engine_github_bot_private_key` | SSH private key of GitHub user with privileges on snapshots and versions repositories. | No default value | ✔︎ |
```

> | `ota_engine_smtp_password` | Password for the SMTP server used for sending errors notifications by email | No default value | - |

Reviewer: ```suggestion
| `ota_engine_smtp_password` | Password for the SMTP server used for sending error notifications by email | No default value | - |
```

| `ota_engine_sendinblue_api_key` | SendInBlue API key used to send email notifications | No default value | - |
| `ota_engine_github_token` | When defined, this authentication token enables GitHub issue creation on the declarations repository | No default value | - |

> | `ota_engine_config_path` | Path to the engine config file related to the inventory file | `../config/production.json` | - |

Reviewer: What does it mean for a config file to be “related” to an inventory file? 🤔

Reviewer: ```suggestion
| `ota_engine_config_path` | Path to the engine config file, relative to the inventory file | `../config/production.json` | - |
```

- - -

## Encrypt sensitive configuration entries

Certain configuration entries contain sensitive information that should be encrypted to ensure security. Ansible provides a convenient way to encrypt such strings using its built-in [vault feature](https://docs.ansible.com/ansible/2.9/user_guide/vault.html):

```sh
ansible-vault encrypt_string --name <sensitive-config-name> <sensitive-config-content>
```

> For example, to encrypt the OTA-Bot private key required by the engine:

Reviewer: ```suggestion
For example, to encrypt the GitHub bot private key used by the engine to push updates:
```

```sh
ansible-vault encrypt_string --name 'ota_engine_ota_bot_private_key' '-----BEGIN OPENSSH PRIVATE KEY-----

> b3BlbnNzaC1rZXktdjEAAAAABG5vbmUAAAAEbm9uZQAAAAAAAAABAAABlwAAAAdzc2gtcn

Reviewer: > I hope that data was forged 😅

Author: Obviously ;)

          66363763613833373530633831653163303961376331393761366261633561656463626563383931
          3361643836623239660a333134626139626465303234313366313433653261376437316231363834
          32643261303534366333383131633430396366343631656363663965633964663331346231663166
          3331316462356461373134303666613035393335333139613639
```

Then it can be used directly in the inventory file:

```yml
all:
  hosts:
    127.0.0.1:
      ansible_user: debian
      ota_engine_config_path: ./engine_config.json
      ota_engine_declarations_branch: new-feature
      ota_engine_ota_bot_private_key: !vault |
          $ANSIBLE_VAULT;1.1;AES256
          62313438616266383732353634343736623532666365643364396464633732613966636235616261
          3136656665316437613434323561613732373361306161640a306132316531356537373862363838
          66363763613833373530633831653163303961376331393761366261633561656463626563383931
          3361643836623239660a333134626139626465303234313366313433653261376437316231363834
          32643261303534366333383131633430396366343631656363663965633964663331346231663166
          3331316462356461373134303666613035393335333139613639
```

> Repeat the process for each sensitive configuration entry that needs encryption.

Reviewer: ```suggestion
Repeat the process for each sensitive configuration entry that needs encryption.

Please note that the data will be stored unencrypted on the deployment server.
```

### Test collection

> _The testing environment is preconfigured for Open Terms Archive maintainers. For other contributors, the configuration file `tests/engine_config.json` needs to be updated to specify repositories where they have authorizations. Additionally, the `ota_engine_github_bot_private_key` value in the inventory file `tests/inventory.yml` should be updated._

Reviewer: > Not sure about the need to use italic 🙂

> Test locally the changes to the collection before opening a pull request:

Reviewer: > Is there a missing command or sentence here? 🤔
